lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -490,15 +502,14 @@ public static function theme_has_support() {
 	 * otherwise returns the whole file path.
 	 *
 	 * @param string $file_name Name of the file.
+	 * @param bool   $template  Use template theme directroy. Default: false.
 	 * @return string The whole file path or empty if the file doesn't exist.
 	 */
-	private static function get_file_path_from_theme( $file_name ) {
-		$located   = '';
-		$candidate = get_theme_file_path( $file_name );
-		if ( is_readable( $candidate ) ) {
-			$located = $candidate;
-		}
-		return $located;
+	private static function get_file_path_from_theme( $file_name, $template = false ) {
+		$path      = $template ? get_template_directory() : get_stylesheet_directory();
+		$candidate = $path . '/' . $file_name;
+
+		return is_readable( $candidate ) ? $candidate : '';
```

Comment:
Can't we simplify this using `get_theme_file_path()`?

```suggestion
	private static function get_file_path_from_theme( $file_name ) {
		$path = get_theme_file_path( $file_name )
		return is_readable( $path ) ? $path : '';
```

Reply:
That's how I initially did it when I introduced basic FSE support for child themes. But now, data is always merged instead of inherited or entirely overridden.

P.S. Your suggestion will return a boolean instead of the path.

Reply:
Ah OK, now I understand what you did 👍 

> P.S. Your suggestion will return a boolean instead of the path.

Oops... you're right. fixed the suggestion and marking this as resolved 👍

Reply:
Thank you, Ari ❤️